.travis.yml

```diff
@@ -42,6 +42,13 @@ matrix:
           packages:
             - *mesa_apt
             - *full_apt
+    - env: PYTHON=2.7 DEPS=full TEST=osmesa IPYTHON="=2" NUMPY="=1.7"
+      addons:
+        apt:
+          sources:
+            - *full_sources
+          packages:
+            - *full_apt
```

Comment:
It's unfortunate that we need to make a Travis run specifically for OSMesa. It would also be nice if one could have both OSMesa and normal GL on a system and switch between them, but maybe that's overkill. It sounds like you've tried all reasonable solutions to get that working, though, so we can live with it. Can you add a comment here saying it needs its own special run for such-and-such reasons?